apps/readest-app/src/pages/api/kosync.ts

```diff
@@ -0,0 +1,49 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { corsAllMethods, runMiddleware } from '@/utils/cors';
+import { KoSyncProxyPayload } from '@/types/kosync';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  await runMiddleware(req, res, corsAllMethods);
+
+  const {
+    serverUrl,
+    endpoint,
+    method,
+    headers: clientHeaders,
+    body: clientBody,
+  } = req.body as KoSyncProxyPayload;
+
+  if (req.method !== 'POST') {
+    return res.status(405).json({ error: 'Method Not Allowed' });
+  }
+
+  if (!serverUrl || !endpoint) {
+    return res.status(400).json({ error: 'serverUrl and endpoint are required' });
+  }
+
+  const targetUrl = `${serverUrl.replace(/\/$/, '')}${endpoint}`;
+
+  try {
+    const response = await fetch(targetUrl, {

```

Comment:
If the `application/vnd.koreader.v1+json` response header is consistently used across all supported koreader-sync implementations, I’d prefer adding this check to prevent potential misuse of the proxy API. Alternatively, we could validate the user authentication token, though that should be considered a last resort.
```
if (response.headers.get('content-type') !== 'application/vnd.koreader.v1+json') {
  throw new Error('Unexpected content type');
}
```